### PR TITLE
chore: improve typings

### PR DESCRIPTION
**Pull request by pf-costa**
The purpose of this pull request is to improve the typings, in order to enforce the integrity of the payloads send to the reducers.

The produced actions returned by useReduction will now the correct typings.

### CODE REVIEW

**Comment by pf-costa** [comment]
@anater thanks for the lib!

File: lib/index.ts [hunk 1]
```diff
@@ -2,15 +2,15 @@ import { useReducer, Reducer, Dispatch, useMemo } from "react";
 
 type Action<T, P> = {
   type: T;
-  payload?: P;
```
Comment (anater) [inline lib/index.ts:5]: Will this cause issues when no payload is provided?

**Comment by anater** [comment]
Thanks for the contribution!

File: lib/index.ts (hunk 1 above)
Comment (pf-costa) [reply lib/index.ts:5]: I've updated the tests in order to allow to trigger an action without specifying the payload. 

test/index.test.ts

**Comment by pf-costa** [comment]
@anater could you kindly merge the PR? I don't have write access.

**Pull request merged**